RevenueCatUI/Templates/V2/Components/Tabs/TabsComponentView.swift

```diff
@@ -106,20 +106,61 @@ struct LoadedTabsComponentView: View {
+        //
+        // This handles a nuanced scenario where tabs may or may not have their own packages:
+        //
+        // Example structure:
+        //   - Package A (parent scope, default)
+        //   - Package B (parent scope)
+        //   - Tabs Component
+        //       - Tab 1: has Package C (its own package)
+        //       - Tab 2: no packages (should inherit from parent)
+        //
+        // Requirements:
+        // 1. Tabs WITH packages: use their own package context, propagate to parent for purchase button
+        // 2. Tabs WITHOUT packages: inherit from parent's selected package
+        // 3. Tabs WITHOUT packages should NOT be affected when tabs WITH packages propagate their package
+        // 4. Tabs WITHOUT packages SHOULD update when user selects a different parent package
+        //
+        // Solution:
+        // - Create a PackageContext for ALL tabs (not just those with packages)
+        // - Tabs with packages: initialized with their own packages
+        // - Tabs without packages: initialized with parent's current state, then kept in sync
+        //   via `onChangeOf` observer that filters out tab propagations (see body)
+        //
         self._tierPackageContexts = .init(initialValue: Dictionary(
-            uniqueKeysWithValues: viewModel.tabViewModels.map { key, tabViewModel in
-                let packageContext = PackageContext(
-                    package: tabViewModel.defaultSelectedPackage,
-                    variableContext: .init(
-                        packages: tabViewModel.packages,
-                        showZeroDecimalPlacePrices: parentPackageContext.variableContext.showZeroDecimalPlacePrices
+            uniqueKeysWithValues: viewModel.tabViewModels.map { key, tabViewModel -> (String, PackageContext) in
+                if !tabViewModel.packages.isEmpty {
+                    // Tab has its own packages - create context with tab's packages
+                    let packageContext = PackageContext(
+                        package: tabViewModel.defaultSelectedPackage,
+                        variableContext: .init(
+                            packages: tabViewModel.packages,
+                            showZeroDecimalPlacePrices: parentPackageContext.variableContext.showZeroDecimalPlacePrices
+                        )
                     )
-                )
-                return (key, packageContext)
+                    return (key, packageContext)
+                } else {
+                    // Tab has no packages - inherit from parent's current state.
+                    // This context will be kept in sync with parent changes via `onChangeOf`
+                    // in the body, which filters out propagations from tabs with packages.
+                    let packageContext = PackageContext(
+                        package: parentPackageContext.package,
+                        variableContext: parentPackageContext.variableContext
+                    )
```

Comment:
Here we're creating a new instance of `PackageContext`, which means that the `onChangeOf` won't really work because the `@Published package` of the new `PackageContext` won't change when the parent's `package` changes. In other words, this requirement
> Tabs WITHOUT packages SHOULD update when user selects a different parent package
would not work right now.

I'm not 100% sure about this, but, if the Tab component doesn't have any package itself at this point (`tabViewModel.packages.isEmpty`), I _think_ you could just do 
```
return (key, parentPackageContext)
```

But we'd need to test it thoroughly, as I'm not sure that this behavior would match the desired behavior (mostly about not propagating the selected package inside Tab 1 to Tab 2)